test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/BodyModelBinderTests.cs

```diff
@@ -116,6 +117,79 @@ public class BodyModelBinderTests
         }
 
         [Fact]
+        public async Task BindModel_NoValueResult_SetsModelStateError()
+        {
+            // Arrange
+            var mockInputFormatter = new Mock<IInputFormatter>();
+            mockInputFormatter.Setup(f => f.CanRead(It.IsAny<InputFormatterContext>()))
+                .Returns(true);
+            mockInputFormatter.Setup(o => o.ReadAsync(It.IsAny<InputFormatterContext>()))
+                              .Returns(InputFormatterResult.NoValueAsync());
```

Comment:
Fixed here and in `BindModel_PassesAllowEmptyInputOptionViaContext`. Scanned the PR and didn't spot any others, but please let me know if you notice any!